DXMainClient/Online/Connection.cs

```diff
@@ -324,6 +326,110 @@ private void HandleComm(object client)
             }
         }
 
+        /// <summary>
+        /// Get all IP addresses of Lobby servers by resolving the hostname and test the latency to the servers.
+        /// The maximum latency is defined in <c>MAXIMUM_LATENCY</c>, see <see cref="Connection.MAXIMUM_LATENCY"/>.
+        /// </summary>
+        /// <returns>A list of available Lobby servers sorted by latency.</returns>
+        private IEnumerable<Server> GetAvailableServerList()
+        {
+            var availableServerAndLatencyDict = new Dictionary<Server, long>();
+
+            try
+            {
+                var dnsAndPingTasks = new List<Task>();
+
+                foreach (var server in Servers)
+                {
+                    var serverHostnameOrIPAddress = server.Host;
+                    var serverName = server.Name;
+                    var serverPorts = server.Ports;
+
+                    var dnsAndPingTask = new Task(() =>
+                    {
+                        Logger.Log($"Attempting to DNS resolve {serverName} ({serverHostnameOrIPAddress}).");
+
+                        // If hostNameOrAddress is an IP address, this address is returned without querying the DNS server.
+                        var serverIPAddresses = Dns.GetHostAddresses(serverHostnameOrIPAddress).Where(item => item.AddressFamily == AddressFamily.InterNetwork).ToList();
+
+                        Logger.Log($"DNS resolved {serverName} ({serverHostnameOrIPAddress}): {string.Join(", ", serverIPAddresses.Select(item => item.ToString()))}");
+
+                        var pingTasks = new List<Task>();
+
+                        foreach (var serverIPAddress in serverIPAddresses)
+                        {
+                            var theIPAddress = serverIPAddress;
```

Comment:
Looks like an absolutely unnecessary local variable.

Reply:
Fixed in 4d3608cc23dc908ee2b838dcfbf9b2b335a52e97 .